File: JAVA/1171. Remove Zero Sum Consecutive Nodes from Linked List.java
class Solution {
    public ListNode removeZeroSumSublists(ListNode head) {
        ListNode dummyHead = new ListNode(0, head);        
        Map<Integer, ListNode> cumulativeSumMap = new HashMap<>();      
        int sum = 0; 
        ListNode current = dummyHead;
        while (current != null) {
            sum += current.val;
            cumulativeSumMap.put(sum, current);
            current = current.next;
        }     
        sum = 0; 
        current = dummyHead;     
        while (current != null) {
            sum += current.val; 
            current.next = cumulativeSumMap.get(sum).next;          
            current = current.next; 
        }      
        return dummyHead.next;
    }
}

File: C++/1171. Remove Zero Sum Consecutive Nodes from Linked List.cpp
class Solution {
public:
    ListNode* removeZeroSumSublists(ListNode* head) {
        ListNode*Head=new ListNode(0);
        Head->next=head;
        ListNode* cur=Head;
        while(cur!=NULL)
        {
            int sum=0;
            cout<<cur->val<<" ";
            while(head!=NULL)
            {
                sum+=head->val;
                if(sum==0)
                {
                   cur->next=head->next; 
                }
                head=head->next;
            }
            cur=cur->next;
            
            if(cur!=NULL){
                head=cur->next;
            }
        }
        return Head->next; 
    }
};
